In [2]:
# Import the modules
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.metrics import confusion_matrix, classification_report

---

## Split the Data into Training and Testing Sets

### Step 1: Read the `lending_data.csv` data from the `Resources` folder into a Pandas DataFrame.

In [3]:
# Read the CSV file from the Resources folder into a Pandas DataFrame
# YOUR CODE HERE!
path = pd.read_csv(Path("Resources/lending_data.csv"))
lending_data_df = pd.DataFrame(path)

# Review the DataFrame
# YOUR CODE HERE!
lending_data_df.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0

### Step 2: Create the labels set (`y`)  from the “loan_status” column, and then create the features (`X`) DataFrame from the remaining columns.

In [4]:
# Separate the data into labels and features

# Separate the y variable, the labels
# YOUR CODE HERE!]
y = lending_data_df["loan_status"].values.reshape(-1,1)

# Separate the X variable, the features
# YOUR CODE HERE!
X = lending_data_df.drop(columns="loan_status", axis=1)

In [5]:
print(type(X))
print(type(y))

<class 'pandas.core.frame.DataFrame'>
<class 'numpy.ndarray'>


In [6]:
# Review the y variable Series
# YOUR CODE HERE!
y

array([[0],
       [0],
       [0],
       ...,
       [1],
       [1],
       [1]], dtype=int64)

In [7]:
# Review the X variable DataFrame
# YOUR CODE HERE!
X.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  
0                 1       22800  
1                 0       13600  
2                 0       16100  
3                 1       22700  
4                 1       23000

### Step 3: Split the data into training and testing datasets by using `train_test_split`.

In [8]:
print(type(X))
print(type(y))

<class 'pandas.core.frame.DataFrame'>
<class 'numpy.ndarray'>


In [9]:
# Import the train_test_learn module
from sklearn.model_selection import train_test_split

# Split the data using train_test_split
# Assign a random_state of 1 to the function
# YOUR CODE HERE!
# X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
# X_train, X_test, y_train, y_test = train_test_split(X, y, )
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42, test_size=0.2)


---

## Create a Logistic Regression Model with the Original Data

###  Step 1: Fit a logistic regression model by using the training data (`X_train` and `y_train`).

In [10]:
# Import the LogisticRegression module from SKLearn
from sklearn.linear_model import LogisticRegression

# Instantiate the Logistic Regression model
# Assign a random_state parameter of 1 to the model
# YOUR CODE HERE!
model = LogisticRegression(random_state=1, max_iter=1000)

# Fit the model using training data
# YOUR CODE HERE!
fit_model = model.fit(X_train, y_train)

c:\Users\jaski\anaconda3\Lib\site-packages\sklearn\utils\validation.py:1143: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


### Step 2: Save the predictions on the testing data labels by using the testing feature data (`X_test`) and the fitted model.

In [11]:
# Make a prediction using the testing data
# YOUR CODE HERE!
test_predictions = model.predict(X_test)
print(len(test_predictions))


15508


### Step 3: Evaluate the model’s performance by doing the following:

* Generate a confusion matrix.

* Print the classification report.

In [13]:
# Generate a confusion matrix for the model
# YOUR CODE HERE!
from sklearn.metrics import confusion_matrix
logic_confuse_matrix = confusion_matrix(y_test, test_predictions)
print(logic_confuse_matrix)

[[14947    64]
 [   47   450]]


In [14]:
# Print the classification report for the model
# YOUR CODE HERE!
logic_class_report = classification_report(y_test, test_predictions)
print(logic_class_report)

              precision    recall  f1-score   support

           0       1.00      1.00      1.00     15011
           1       0.88      0.91      0.89       497

    accuracy                           0.99     15508
   macro avg       0.94      0.95      0.94     15508
weighted avg       0.99      0.99      0.99     15508



### Step 4: Answer the following question.

**Question:** How well does the logistic regression model predict both the `0` (healthy loan) and `1` (high-risk loan) labels?

**Answer:** WRITE YOUR ANSWER HERE!
The Logistical Regression Model predicts a healthy loan with 100% precision and 99% were able to be captured. The model predicted the high risk loan (1) to have 88% precision, this means 12% of the time the model could falsley predict a high risk loan.

---